src/Core/src/Hosting/ImageSources/ImageSourceServiceProvider.cs

```diff
@@ -22,40 +25,59 @@ public ImageSourceServiceProvider(IImageSourceServiceCollection collection, ISer
 
 		public IServiceProvider HostServiceProvider { get; }
 
-		public IImageSourceService? GetImageSourceService(Type imageSource) =>
+		public IImageSourceService? GetImageSourceService([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type imageSource) =>
 			(IImageSourceService?)GetService(GetImageSourceServiceType(imageSource));
 
-		public Type GetImageSourceServiceType(Type imageSource) =>
-			_serviceCache.GetOrAdd(imageSource, type =>
-			{
-				var genericConcreteType = ImageSourceServiceType.MakeGenericType(type);
+		[UnconditionalSuppressMessage("Trimming", "IL2073", Justification = TrimmerHelper.ConcurrentDictionary)]
+		[UnconditionalSuppressMessage("Trimming", "IL2111", Justification = TrimmerHelper.MakeGenericType)]
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+		public Type GetImageSourceServiceType([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type imageSource) =>
+			_serviceCache.GetOrAdd(imageSource, CreateImageSourceServiceType);
 
-				if (genericConcreteType != null && GetServiceDescriptor(genericConcreteType) != null)
-					return genericConcreteType;
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+		Type CreateImageSourceServiceType([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type type)
+		{
+			var genericConcreteType = MakeGenericType(ImageSourceServiceType, type);
 
-				return ImageSourceServiceType.MakeGenericType(GetImageSourceType(type));
-			});
+			if (genericConcreteType != null && GetServiceDescriptor(genericConcreteType) != null)
+				return genericConcreteType;
 
-		public Type GetImageSourceType(Type imageSource) =>
+			return MakeGenericType(ImageSourceServiceType, GetImageSourceType(type));
+		}
+
+		[UnconditionalSuppressMessage("Trimming", "IL2073", Justification = TrimmerHelper.ConcurrentDictionary)]
+		[UnconditionalSuppressMessage("Trimming", "IL2111", Justification = TrimmerHelper.MakeGenericType)]
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+		public Type GetImageSourceType([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type imageSource) =>
 			_imageSourceCache.GetOrAdd(imageSource, CreateImageSourceTypeCacheEntry);
 
-		Type CreateImageSourceTypeCacheEntry(Type type)
+		[UnconditionalSuppressMessage("Trimming", "IL2065", Justification = "'directInterface.GetInterface(fullName)' can not be statically determined and may not meet 'DynamicallyAccessedMembersAttribute' requirements.")]
```

Comment:
This is worth a comment - it's basically because the trimmer can't track annotations through collections. The suppression is correct, but I think it's better to describe exactly why - going forward if the method body changes, the suppression may suppress other issues - the command would at least have some hint as to why it's there.